# RNA-seq data analysis

Many resources and introductory tutorials to typical RNA-seq workflows exist on the web. The choice of tools and computational methods is often a matter of choice. Generally R/Bioconductor is a great resource for HTS-data related tutorials.

Often the best approach is to locally reproduce results discussed as part of different tools. For example, most R/Bioconductor packages provide sample analyses and case studies as part of their vignettes.   

## General workflow

A typical RNA-seq data analysis workflow may consist of (some or all of) the following steps (in brackets are tools that I have used in the past):

1. Quality assessment of raw reads ([fastqc](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/))
2. Adapter trimming and quality filtering ([cutadapt](http://cutadapt.readthedocs.io/en/stable/guide.html), [trimmomatic](http://www.usadellab.org/cms/?page=trimmomatic), [bbduk.sh](https://sourceforge.net/projects/bbmap/))
3. Read alignment to reference genome ([bowtie2](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml), [tophat](https://ccb.jhu.edu/software/tophat/index.shtml), [bwa-mem](http://bio-bwa.sourceforge.net/), [STAR](https://github.com/alexdobin/STAR))
4. Quality assessment of mapped reads ([RseQC](http://rseqc.sourceforge.net/), [QoRTs](https://github.com/hartleys/QoRTs))
5. Read de-duplication ([samtools](http://www.htslib.org/), [picard-tools](https://broadinstitute.github.io/picard/))
6. Count summarisation ([HTSeq](http://www-huber.embl.de/users/anders/HTSeq/), [featureCounts](http://bioinf.wehi.edu.au/featureCounts/))
7. Differential gene expression analysis ([DESeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html), [limma](https://bioconductor.org/packages/release/bioc/html/limma.html), [edgeR](http://bioconductor.org/packages/release/bioc/html/edgeR.html))
8. Gene ontology enrichment analysis/Gene set enrichment analysis ([clusterProfiler](https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html), [GSEA](http://software.broadinstitute.org/gsea/index.jsp))

---

[Conesa et al.](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8) have published a survey of best practices for RNA-seq data analysis that provides a nice overview of typical analysis steps and how to critically assess results.


---

[Griffith lab RNA-seq tutorial](https://github.com/griffithlab/rnaseq_tutorial/wiki)

Griffith et al. provide an online Wiki-style RNA-seq tutorial with exercises.

## Differential gene expression (DGE) analysis

I can specifically recommend the following three tutorials/vignettes surrounding the R packages DESeq2 and limma. In terms of the underlying statistical model, there exists a lot of overlap between DESeq2 and limma. Whether to use one over the other is therefore largely a matter of choice. Both methods are often regarded as the gold-standard methods in regards to DGE analyses.

---

[http://www.bioconductor.org/help/workflows/rnaseqGene/](http://www.bioconductor.org/help/workflows/rnaseqGene/)

R/Bioconductor has a good tutorial on how to get from raw RNA-seq data to identifying differentially expressed genes.


---

[DESeq2](https://www.bioconductor.org/packages/release/bioc/html/DESeq2.html)

The vignette is regularly updated and offers a great tutorial on how to perform differential gene expression analyses within R. The vignette also provides details on the underlying statistical modelling approach. Personally, I slightly favour the DESeq2 documentation and approach in my own research analyses.

---

[Limma](https://www.bioconductor.org/packages/release/bioc/html/limma.html)

The user guide offers an excellent tutorial using a wide range of different sample case studies on performing differential gene expression analyses within R.


## Fusion gene identification

Various fusion/chimeric transcript identification tools exist.

---
[Kumar et al.](http://www.nature.com/articles/srep21597) performed a quantitative assessment of 12 fusion detection tools in their recent 2016 Nature paper.

---

[Liu et al.](http://nar.oxfordjournals.org/content/44/5/e47) performed a similar benchmark analysis of 15 fusion detection tools in their 2015 Nucleic Acids Research paper.

---

Another list of 39 (as of December 2016) fusion detection tools in given on [Biostars](https://www.biostars.org/p/45986/).  

---

[InFusion](https://bitbucket.org/kokonech/infusion/overview)

InFusion is one of the most recently published tools. It was written in C++/Python, and provides detailed instructions on how to compile the source code, and run the program.

---
