<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>High-throughput sequencing data analysis</title>
  <meta name="description" content="High-throughput sequencing data analysis">
  <meta name="generator" content="bookdown 0.4 and GitBook 2.6.7">

  <meta property="og:title" content="High-throughput sequencing data analysis" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="High-throughput sequencing data analysis" />
  
  
  

<meta name="author" content="Maurits Evers">


<meta name="date" content="2018-01-30">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="rna-seq-data-analysis.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="general-statistics-and-statistical-modelling.html"><a href="general-statistics-and-statistical-modelling.html"><i class="fa fa-check"></i><b>1</b> General statistics and statistical modelling</a><ul>
<li class="chapter" data-level="1.1" data-path="general-statistics-and-statistical-modelling.html"><a href="general-statistics-and-statistical-modelling.html#books"><i class="fa fa-check"></i><b>1.1</b> Books</a></li>
<li class="chapter" data-level="1.2" data-path="general-statistics-and-statistical-modelling.html"><a href="general-statistics-and-statistical-modelling.html#tutorial-and-notes"><i class="fa fa-check"></i><b>1.2</b> Tutorial and notes</a></li>
<li class="chapter" data-level="1.3" data-path="general-statistics-and-statistical-modelling.html"><a href="general-statistics-and-statistical-modelling.html#free-ebooks"><i class="fa fa-check"></i><b>1.3</b> Free ebooks</a></li>
<li class="chapter" data-level="1.4" data-path="general-statistics-and-statistical-modelling.html"><a href="general-statistics-and-statistical-modelling.html#web-sources-blogs-and-online-forums"><i class="fa fa-check"></i><b>1.4</b> Web sources, blogs and online forums</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="general-bioinformatics.html"><a href="general-bioinformatics.html"><i class="fa fa-check"></i><b>2</b> General bioinformatics</a><ul>
<li class="chapter" data-level="2.1" data-path="general-bioinformatics.html"><a href="general-bioinformatics.html#books-1"><i class="fa fa-check"></i><b>2.1</b> Books</a></li>
<li class="chapter" data-level="2.2" data-path="general-bioinformatics.html"><a href="general-bioinformatics.html#free-ebooks-notes-and-manuscripts"><i class="fa fa-check"></i><b>2.2</b> Free Ebooks, notes, and manuscripts</a></li>
<li class="chapter" data-level="2.3" data-path="general-bioinformatics.html"><a href="general-bioinformatics.html#web-sources-blogs-and-online-forums-1"><i class="fa fa-check"></i><b>2.3</b> Web sources, blogs and online forums</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="r-programming.html"><a href="r-programming.html"><i class="fa fa-check"></i><b>3</b> R programming</a><ul>
<li class="chapter" data-level="3.1" data-path="r-programming.html"><a href="r-programming.html#tutorial-and-notes-1"><i class="fa fa-check"></i><b>3.1</b> Tutorial and notes</a></li>
<li class="chapter" data-level="3.2" data-path="r-programming.html"><a href="r-programming.html#ebooks"><i class="fa fa-check"></i><b>3.2</b> Ebooks</a></li>
<li class="chapter" data-level="3.3" data-path="r-programming.html"><a href="r-programming.html#online-courses"><i class="fa fa-check"></i><b>3.3</b> Online courses</a></li>
<li class="chapter" data-level="3.4" data-path="r-programming.html"><a href="r-programming.html#web-sources-blogs-and-online-forums-2"><i class="fa fa-check"></i><b>3.4</b> Web sources, blogs and online forums</a></li>
<li class="chapter" data-level="3.5" data-path="r-programming.html"><a href="r-programming.html#specialised-topics"><i class="fa fa-check"></i><b>3.5</b> Specialised topics</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="rna-seq-data-analysis.html"><a href="rna-seq-data-analysis.html"><i class="fa fa-check"></i><b>4</b> RNA-seq data analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="rna-seq-data-analysis.html"><a href="rna-seq-data-analysis.html#general-workflow"><i class="fa fa-check"></i><b>4.1</b> General workflow</a></li>
<li class="chapter" data-level="4.2" data-path="rna-seq-data-analysis.html"><a href="rna-seq-data-analysis.html#differential-gene-expression-dge-analysis"><i class="fa fa-check"></i><b>4.2</b> Differential gene expression (DGE) analysis</a></li>
<li class="chapter" data-level="4.3" data-path="rna-seq-data-analysis.html"><a href="rna-seq-data-analysis.html#fusion-gene-identification"><i class="fa fa-check"></i><b>4.3</b> Fusion gene identification</a></li>
<li class="chapter" data-level="4.4" data-path="rna-seq-data-analysis.html"><a href="rna-seq-data-analysis.html#read-simulation"><i class="fa fa-check"></i><b>4.4</b> Read simulation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chip-seq-data-analysis.html"><a href="chip-seq-data-analysis.html"><i class="fa fa-check"></i><b>5</b> ChIP-seq data analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="chip-seq-data-analysis.html"><a href="chip-seq-data-analysis.html#general-workflow-1"><i class="fa fa-check"></i><b>5.1</b> General workflow</a></li>
<li class="chapter" data-level="5.2" data-path="chip-seq-data-analysis.html"><a href="chip-seq-data-analysis.html#online-resources"><i class="fa fa-check"></i><b>5.2</b> Online resources</a></li>
<li class="chapter" data-level="5.3" data-path="chip-seq-data-analysis.html"><a href="chip-seq-data-analysis.html#the-cistrome-project"><i class="fa fa-check"></i><b>5.3</b> The Cistrome project</a></li>
<li class="chapter" data-level="5.4" data-path="chip-seq-data-analysis.html"><a href="chip-seq-data-analysis.html#controls-and-replicate-libraries"><i class="fa fa-check"></i><b>5.4</b> Controls and replicate libraries</a></li>
<li class="chapter" data-level="5.5" data-path="chip-seq-data-analysis.html"><a href="chip-seq-data-analysis.html#library-quality-assessment"><i class="fa fa-check"></i><b>5.5</b> Library quality assessment</a><ul>
<li class="chapter" data-level="5.5.1" data-path="chip-seq-data-analysis.html"><a href="chip-seq-data-analysis.html#alignment-stats"><i class="fa fa-check"></i><b>5.5.1</b> Alignment stats</a></li>
<li class="chapter" data-level="5.5.2" data-path="chip-seq-data-analysis.html"><a href="chip-seq-data-analysis.html#read-duplication"><i class="fa fa-check"></i><b>5.5.2</b> Read duplication</a></li>
<li class="chapter" data-level="5.5.3" data-path="chip-seq-data-analysis.html"><a href="chip-seq-data-analysis.html#fingerprint-plot"><i class="fa fa-check"></i><b>5.5.3</b> Fingerprint plot</a></li>
<li class="chapter" data-level="5.5.4" data-path="chip-seq-data-analysis.html"><a href="chip-seq-data-analysis.html#further-qc"><i class="fa fa-check"></i><b>5.5.4</b> Further QC</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="chip-seq-data-analysis.html"><a href="chip-seq-data-analysis.html#references"><i class="fa fa-check"></i><b>5.6</b> References</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">High-throughput sequencing data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chip-seq-data-analysis" class="section level1">
<h1><span class="header-section-number">5</span> ChIP-seq data analysis</h1>
<p>Work in progress (January 2018)…</p>
<div id="general-workflow-1" class="section level2">
<h2><span class="header-section-number">5.1</span> General workflow</h2>
<p>A typical workflow for the analysis of transcription factor (TF) ChIP-seq data may consist of the following steps. Tools are given in brackets.</p>
<ol style="list-style-type: decimal">
<li>Quality assessment of raw reads (<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a>)</li>
<li>Adapter trimming and quality filtering (<a href="http://cutadapt.readthedocs.io/en/stable/guide.html">cutadapt</a>, <a href="http://www.usadellab.org/cms/?page=trimmomatic">trimmomatic</a>, <a href="https://sourceforge.net/projects/bbmap/">bbduk.sh</a>)</li>
<li>Read alignment to reference genome (<a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">bowtie2</a>)</li>
<li>Quality assessment of mapped reads (<a href="http://qualimap.bioinfo.cipf.es/">QualiMap</a>)</li>
<li>Read de-duplication (<a href="http://www.htslib.org/">samtools</a>, <a href="https://broadinstitute.github.io/picard/">picard-tools</a>)</li>
<li>ChIP library assessment (<a href="https://deeptools.github.io/">deepTools</a>)</li>
<li>Meta-gene analysis (<a href="https://deeptools.github.io/">deepTools</a>)</li>
<li>Peak calling (<a href="https://github.com/taoliu/MACS">MACS2</a>)</li>
</ol>
</div>
<div id="online-resources" class="section level2">
<h2><span class="header-section-number">5.2</span> Online resources</h2>
<ol style="list-style-type: decimal">
<li><p><a href="http://ged.msu.edu/angus/tutorials-2011/files/lecture-chipseq.pdf">ChIP-seq Analysis, Albert, Bioinformatics Consulting Center @ Penn State University</a></p></li>
<li><p><a href="http://biocluster.ucr.edu/~rkaundal/workshops/R_feb2016/ChIPseq/ChIPseq.html">NHS data analysis with R/Bioconductor: ChIP-seq workflow, Kaundal, UC Riverside</a></p></li>
<li><p><a href="http://bioinformatics-core-shared-training.github.io/ngs-in-bioc/Lectures/Lect6b_ChIP-Seq%20Data%20Analysis.pdf">ChIP-seq Data Analysis, Menon, Bioinformatics core @ CRUK Cambridge Institute, University of Cambridge</a></p></li>
<li><p><a href="http://cbsu.tc.cornell.edu/lab/doc/CHIPseq_workshop_20150504_lecture1.pdf">Computational pipeline for ChIP-seq Data Analysis, Wang and Sun, Institute of Biotechnology, Cornell University</a></p></li>
</ol>
</div>
<div id="the-cistrome-project" class="section level2">
<h2><span class="header-section-number">5.3</span> The Cistrome project</h2>
<blockquote>
<p>The cistrome refers to “the set of cis-acting targets of a trans-acting factor on a genome-wide scale, also known as the in vivo genome-wide location of transcription factor binding-sites or histone modifications”.</p>
</blockquote>
<p>The <a href="http://www.cistrome.org/Cistrome/Cistrome_Project.html">cistrome project</a> includes a variety of ChIP-seq data analysis tools developed by the lab of <a href="http://liulab.dfci.harvard.edu/">Shirley Liu</a>.</p>
<p>For example, the <a href="http://cistrome.org/db/#/">Cistrome Data Browser</a> is a good resource for identifying publicly available ChIP-seq and DNase-seq data.</p>
</div>
<div id="controls-and-replicate-libraries" class="section level2">
<h2><span class="header-section-number">5.4</span> Controls and replicate libraries</h2>
<p>It is critical to have controls for the ChIP libraries to estimate non-specific binding across the genome. The two most frequently used methods use (1) DNA input samples, or (2) IgG “mock-ChIP” control samples. Various publications discuss potential benefits of one method over the other, and it appears that the choice of controls is to some extent based on individual experience and personal preference. Overall, the use of input DNA as controls seems to be preferred over IgG controls (see e.g. <a href="http://www.nature.com/ni/journal/v12/n10/abs/ni.2117.html">Kiddler et al.</a>, <a href="http://genome.cshlp.org/content/22/9/1813.long">Landt et al.</a>, <a href="http://www.nature.com/nrg/journal/v15/n11/full/nrg3788.html">Meyer and Liu</a>). It is important to keep in mind that control libraries are subject to systematic biases due to genomic copy number variations, sequencing biases, read mapping ambiguities, and cell-type specific chromatin structure <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0005241">[Vega et al.]</a>.</p>
<p>The original 2012 ENCODE guidelines <a href="http://genome.cshlp.org/content/22/9/1813.long">[Land et al.]</a> state that “more than two replicates did not significantly improve site discovery”, and therefore define a standard for ChIP measurements to “be performed on two independent biological repliates”. Results from more recent publications strongly encourage <span class="math inline">\(n\geq 3\)</span> replicate libraries to increase the reliability in the identification of binding sites. For example, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3962196/">[Yang et al.]</a> demonstrate that a simple majority-rule-approach “identifies peaks more reliably in all biological replicates than the absolute concordance of peak identification between any two replicates, further demonstrating the utility of increasing replicate numbers in ChIP-seq experiments”.</p>
</div>
<div id="library-quality-assessment" class="section level2">
<h2><span class="header-section-number">5.5</span> Library quality assessment</h2>
<div id="alignment-stats" class="section level3">
<h3><span class="header-section-number">5.5.1</span> Alignment stats</h3>
<p>Following read alignment, <code>bowtie2</code>’s alignment summaries give a first insight into library quality and sequencing depth. Using an unmasked reference genome, mappability of input samples is commonly around 70-90%. In a repeat-masked reference genome, around 50% of the genomic sequence is identified as interspersed repeats and low-complexity DNA and masked with N’s (see e.g. <a href="http://www.repeatmasker.org/">RepeatMasker</a>); consequently, alignment of input reads against a repeat-masked reference genome will give lower mappability rates around 35-55%. For example, we use a repeat-masked mouse reference genome with an additional copy of the canonical rDNA repeat unit to explore protein binding to the rDNA, and obtain mappability rates of around 40% for high-quality input samples.</p>
</div>
<div id="read-duplication" class="section level3">
<h3><span class="header-section-number">5.5.2</span> Read duplication</h3>
<p>Generally, quality control of aligned reads should include an assessment of the duplication percentage of read (pairs). It is generally advisable to remove duplicate read pairs prior to proceeding with any down-stream analysis; for example, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4597324/">Dozmorov et al.</a> conclude that</p>
<blockquote>
<p>[b]oth adapter trimming and duplicate removal moderately improved the strength of biological signals in RNA-seq and ChIP-seq data. Aggressive filtering of reads overlapping with low complexity regions, as defined by RepeatMasker, further improved the strength of biological signals, and the correlation between RNA-seq and microarray gene expression data.</p>
</blockquote>
<p>Some peak-callers (like MACS2) will indepedently (and by default) remove read duplicates as part of their analysis workflow.</p>
<p>We can identify and mark duplicate reads using <code>picard-tools MarkDuplicates</code> and/or remove duplicates directly using <code>picard-tools MarkDuplicates REMOVE_DUPLICATES=TRUE</code>. Alternatively, we can use <code>samtools rmdup</code> to remove duplicates. The total percentage of duplicate reads and their chromosomal distribution provide insight into library complexity and biases.</p>
<p>Additional notes on <code>samtools rmdup</code> vs. <code>picard-tools MarkDuplicates</code> taken from a <a href="https://gatkforums.broadinstitute.org/gatk/discussion/6793/picard-markduplicates-vs-samtools-rmdup-for-variant-calling-with-gatk">discussion on the GATK forum</a>:</p>
<ul>
<li><p><code>MarkDuplicates</code> by default marks properly paired PE reads and SE reads (e.g. mate unmapped but not secondary/supplementary) with the <code>1024</code> flag (hexadecimal: <code>0x400</code>) <strong>without removing the read</strong>. <code>MarkDuplicates</code> scores primary mapped reads based on the sum of base quality scores for both mates of a pair.</p></li>
<li><p><code>rmdup</code> retains the read pair with the highest mapping quality, and <strong>removes any duplicate read pairs</strong> from the output BAM/SAM file.</p></li>
</ul>
</div>
<div id="fingerprint-plot" class="section level3">
<h3><span class="header-section-number">5.5.3</span> Fingerprint plot</h3>
<p>A powerful tool to assess the quality of libraries from a ChIP-seq experiment is to plot the <em>fingerprint</em> of a library. The term “fingerprint (plot)” has been coined by the deepTools dev’s, but the original idea was introduced by <a href="https://github.com/songlab/chance/wiki/CHANCE-Manual#checking-the-strength-of-enrichment-in-the-ip">Diaz et al.</a></p>
<p>From <a href="http://deeptools.readthedocs.io/en/latest/content/tools/plotFingerprint.html">deepTools</a>:</p>
<blockquote>
<p>This quality control will most likely be of interest for you if you are dealing with ChIP-seq samples as a pressing question in ChIP-seq experiments is “Did my ChIP work?”, i.e. did the antibody-treatment enrich sufficiently so that the ChIP signal can be separated from the background signal? (After all, around 90% of all DNA fragments in a ChIP experiment will represent the genomic background).</p>
</blockquote>
<p>The fingerprint of a ChIP/input library characterises how much of the genome (that is covered by reads) is covered by how large a fraction of the total number of reads. The following figure is taken from <a href="http://deeptools.readthedocs.io/en/latest/content/tools/plotFingerprint.html">deepTools’ documentation</a>. <img src="ChIPseq_fingerprint.png" alt="ChIPseq_fingerprint" style="height: 400px;"/></p>
A simple example on how the fingerprint distribution is calculated from a read distribution is shown in the following diagram.
<p style="text-align:center; margin-top:1em">
<img src="example_fingerprint.png" alt="example_fingerprint" style="height: 300px;"/>
</p>
<p><strong>Detour</strong>: Fingerprint plots are closely related to <a href="https://en.wikipedia.org/wiki/Lorenz_curve">Lorenz curves</a> in economics. Usually in a Lorenz curve the fraction of a population is plotted against the fraction of total income, and they are a useful tool for assessing income inequality. If income is perfectly equally distributed, the Lorenz curve will be a straight line (just like an equal non-specific read distribution across the genome will give rise to a straight line with zero offset and unit slope). In the context of Lorenz curves, it is common to summarise income inequality in terms of the <a href="https://en.wikipedia.org/wiki/Gini_coefficient">Gini coefficient</a>: the larger the Gini coefficient, the larger the income inequality. It is interesting to note that – to my knowledge – the close connection between fingerprint plots and Lorenz curves has not been discussed. In ChIP-seq fingerprint plots, the corresponding Gini coefficient could be used as a critical summary statistic, characterising the degree of specific binding, with a zero Gini coefficient indicating completely uniform non-specific background binding.</p>
</div>
<div id="further-qc" class="section level3">
<h3><span class="header-section-number">5.5.4</span> Further QC</h3>
<p>Further quality control checks typically involve a principle component analysis of read coverage (1) across the full (binned) genome, and (2) within identified common peak regions. Following appropriate centering of read counts, samples should separate along the first principle component according to whether they are input or ChIP libraries. Separation on the second principle component should then occur according to e.g. biological replicates, time-points or ChIP antibody (depending on the experiment design).</p>
<p>Additionally, a correlation analysis based on either counts directly or on their ranks gives a summary statistic characterising within-replicate reproducibility. It is important to keep in mind that Pearson’s product moment correlation coefficient will dependent on the dynamic range of the two input variables; we therefore expect on average larger correlation coefficients in pairwise comparisons of input libraries than of ChIP libraries.</p>
</div>
</div>
<div id="references" class="section level2">
<h2><span class="header-section-number">5.6</span> References</h2>
<ol style="list-style-type: decimal">
<li><a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003326">Bailey et al., Practical Guidelines for the Comprehensive Analysis of ChIP-seq Data, PLoS Comp. Biol. 9, e1003326 (2013)</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4597324/">Dozmorov et al., Detrimental effects of duplicate reads and low complexity regions on RNA- and ChIP-seq data, BMC Bioinformatics 16, S10 (2015)</a></li>
<li><a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-12-134">Ho et al., ChIP-chip versus ChIP-seq: Lessons for experimental design and data analysis, BMC Genomics 12, 134 (2011)</a></li>
<li><a href="http://www.nature.com/ni/journal/v12/n10/abs/ni.2117.html">Kidder, Hu and Zhao, ChIP-Seq: technical considerations for obtaining high-quality data, Nature Immunology 12, 918 (2011)</a></li>
<li><a href="http://genome.cshlp.org/content/22/9/1813.long">Landt et al., ChIP-seq guidelines and practices of the ENCODE and modENCODE consortia, Genome R5esearch 22, 1813 (2012)</a></li>
<li><a href="http://www.nature.com/nrg/journal/v15/n11/full/nrg3788.html">Meyer and Liu, Identifying and mitigating bias in next-generation sequencing methods for chromatin biology, Nature Reviews Genetics 15, 709 (2014)</a></li>
<li><a href="https://academic.oup.com/bib/article/18/2/279/2453282/Recent-advances-in-ChIP-seq-analysis-from-quality">Nakato and Shirahige, Recent advances in ChIP-seq analysis: from quality management to whole-genome annotation, Briefings in Bioinformatics 18, 279 (2017)</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4574076/">Ramachandran, Palidwor and Perkins, BIDCHIPS: bias decomposition and removal from ChIP-seq data clarifies true binding signal and its functional correlates, Epigenetics Chromatin 8, 33 (2015)</a></li>
<li><a href="http://www.biorxiv.org/content/early/2017/01/15/090704">Teng and Irizarry, Accounting for GC-content bias reduces systematic errors and batch effects in ChIP-Seq data, bioRxiv 10.1101/090704 (2017)</a></li>
<li><a href="https://academic.oup.com/bib/article/18/3/441/2453291/Features-that-define-the-best-ChIP-seq-peak">Thomas et al., Features that define the best ChIP-seq peak calling algorithms, Briefings in Bioinformatics 18, 441 (2017)</a></li>
<li><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0005241">Vega et al., Inherent Signals in Sequencing-Based Chromatin-ImmunoPrecipitation Control Libraries, PLoS ONE 4, e5241 (2009)</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3962196/">Yang et al., Leveraging biological replicates to improve analysis in ChIP-seq experiments, Comput. Struct. Biotechnol. J. 9, e201401002 (2014)</a></li>
</ol>
<p>More web resources</p>
<p>Notwithstanding the username, <a href="https://github.com/crazyhottommy/ChIP-seq-analysis">crazyhottommy</a> lists many resources/links associated with ChIP-seq data analysis.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="rna-seq-data-analysis.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
